numba/core/typing/builtins.py

```diff
@@ -995,6 +995,16 @@ def generic(self, args, kws):
 
         [arg] = args
 
+        if isinstance(arg, types.UnicodeType):
+            msg = 'argument must be a string literal'
```

Comment:
The argument could be a number, it needn't be a string literal. Perhaps:

```suggestion
            msg = 'string argument must be a literal'
```

reflects the issue more accurately?

Also - I think this branch is untested.